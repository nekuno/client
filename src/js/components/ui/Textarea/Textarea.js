import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Textarea.scss';

export default class Textarea extends Component {

    static propTypes = {
        placeholder : PropTypes.string,
        defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        onChange    : PropTypes.func
    };

    constructor(props) {
        super(props);

        this.state = {
            empty: !props.defaultValue,
            focused: false
        };

        this.onChange = this.onChange.bind(this);
        this.getValue = this.getValue.bind(this);
    }

    setValue(value) {
        if (this.refs.textarea) {
            this.refs.textarea.value = value;
        }
    }

    getValue() {
        return this.refs.textarea ? this.refs.textarea.value : null;
    }

    onChange() {
        if (this.props.onChange) {
            this.props.onChange(this.getValue());
        }

        this.setState({empty: !this.getValue()});
    }

    render() {
        const {placeholder, defaultValue} = this.props;
        const {empty} = this.state;


        return (
            <div className={styles.inputWrapper}>
                <div className={styles.input}>
                    <textarea ref="textarea" placeholder={placeholder} value={this.state.value} onChange={this.onChange} defaultValue={defaultValue} />
                </div>
            </div>
        );
    }
}