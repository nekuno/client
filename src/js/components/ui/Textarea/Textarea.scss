@import '../../../../scss/_partials/colors';

.input-wrapper {
  padding: 8px;

  .input {
    border-radius: 32px;
    border: 1px solid $border;
    color: $gray-main-text;
    height: 200px;
    line-height: 60px;
    margin: 16px;
    text-align: center;

    textarea {
      background-color: $background-white;
      border: none;
      -moz-box-shadow: none;
      -webkit-box-shadow: none;
      box-shadow: none;
      font-family: Arial;
      font-size: 16px;
      height: 200px;
      line-height: 40px;
      overflow: auto;
      text-align: left;
      width: 85%;
      &:focus {
        outline: none;
      }

      .icon-cancel {
        cursor: pointer;
      }
      .icon-check {
        color: $green-experiences
      }
    }
  }
}

