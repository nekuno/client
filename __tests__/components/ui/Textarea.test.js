import React from 'react';
import Textarea from '../../../src/js/components/ui/Textarea/Textarea.js';
import styles from '../../../src/js/components/ui/Textarea/Textarea.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Textarea component', () => {
    const mockFn = jest.fn();
    const textarea = shallow(<Textarea onClickHandler={mockFn}>Lorem ipsum</Textarea>);

    it('should be defined', () => {
        expect(textarea).toBeDefined();
    });
    it('should render correctly', () => {
        expect(textarea).toMatchSnapshot();
    });
});
