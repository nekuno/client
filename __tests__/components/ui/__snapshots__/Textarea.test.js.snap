// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Textarea component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Textarea
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </Textarea>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="input"
      >
        <textarea
          onChange={[Function]}
        />
      </div>,
      "className": "inputWrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <textarea
          onChange={[Function]}
        />,
        "className": "input",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "defaultValue": undefined,
          "onChange": [Function],
          "placeholder": undefined,
          "value": undefined,
        },
        "ref": "textarea",
        "rendered": null,
        "type": "textarea",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="input"
        >
          <textarea
            onChange={[Function]}
          />
        </div>,
        "className": "inputWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <textarea
            onChange={[Function]}
          />,
          "className": "input",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": undefined,
            "onChange": [Function],
            "placeholder": undefined,
            "value": undefined,
          },
          "ref": "textarea",
          "rendered": null,
          "type": "textarea",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
